include/oneapi/dpl/internal/dynamic_selection_impl/submission_group.h

```diff
@@ -18,31 +18,29 @@ namespace dpl {
 namespace experimental {
   template<typename T>
   class submission_group_t {
-    using submission_container_t = util::concurrent_queue<T>;
+    using submission_container_t = std::shared_ptr<util::concurrent_queue<T>>;
     using submissions_t = std::vector<T>;
     submission_container_t submission_container;
     submissions_t submissions;
 
     public:
-    submission_group_t() = default;
-    submission_group_t(const submission_group_t&) = delete;
-    submission_group_t& operator=(const submission_group_t&) = delete;
+    submission_group_t() : submission_container(std::make_shared<util::concurrent_queue<T>>()) {};
 
     //add to submission group after submit
-    void add_submission(T w) { 
-      submission_container.push_back(w);	
-    }	
+    void add_submission(T w) {
+      submission_container->push(w);
+    }
 
     //wait on all submissions in the list
     void wait() {
-      while(!submission_container.empty()){
-        T w;  
-        submission_container.pop(w);
+      while(!submission_container->empty()){
+        T w;
+        submission_container->pop(w);
         w->wait();
         delete w;
       }
     }
```

Comment:
If this function is supposed to be used by multiple threads, it is incorrect. Different threads will pop different elements from the queue and therefore will wait only on a part of submissions.